test/gtest/uct/test_md.cc

```diff
@@ -106,6 +113,64 @@ void test_md::check_caps(uint64_t flags, const std::string& name)
     }
 }
 
+void test_md::alloc_memory(void **address, size_t size, int fill, int mem_type)
+{
+    if (mem_type == UCT_MD_MEM_TYPE_HOST) {
+        *address = malloc(size);
+        ASSERT_TRUE(*address != NULL);
+        memset(*address, fill, size);
+    } else if (mem_type == UCT_MD_MEM_TYPE_CUDA) {
+#if HAVE_CUDA
+        void *temp;
+        cudaError_t ret;
+
+        ret = cudaMalloc(address, size);
+        ASSERT_TRUE(ret == cudaSuccess);
+
+        temp = malloc(size);
+        ASSERT_TRUE(temp != NULL);
+        memset(temp, fill, size);
+        ret = cudaMemcpy(*address, temp, size, cudaMemcpyHostToDevice);
+        ASSERT_TRUE(ret == cudaSuccess);
+        free(temp);
+#else
+        std::stringstream ss;
+        ss << "can't allocate cuda memory for " << GetParam();
+        UCS_TEST_SKIP_R(ss.str());
+#endif
+    }
+}
+
+void test_md::check_memory(void *address, size_t pos, int expect, int mem_type)
+{
+    if (mem_type == UCT_MD_MEM_TYPE_HOST) {
+        EXPECT_EQ(expect, *((char*)address + pos));
+    } else if (mem_type == UCT_MD_MEM_TYPE_CUDA) {
+#if HAVE_CUDA
+        void *temp;
+        cudaError_t ret;
+
+        temp = malloc(1);
```

Comment:
why not check the entire buffer?

Reply:
Just wanted to keep the same  checking as host buffer case where it is checking last byte content

Reply:
so let's check all buffer in both cases
for example use ucs::fill_random() to generate random data

Reply:
fixed